packages/core/src/common/unsafe_abstractComponent.ts

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2015 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the terms of the LICENSE file distributed with this project.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as React from "react";
+import { isNodeEnv } from "./utils";
+
+/**
+ * An abstract component that Blueprint components can extend
+ * in order to add some common functionality like runtime props validation.
+ * This version will not work with React 17 upwards and only exists for legacy purposes
+ */
+/* tslint:disable-next-line:class-name */
+export abstract class UNSAFE_AbstractComponent<P, S> extends React.Component<P, S> {
```

Comment:
this isn't exactly what I was hoping for... in order to make this change without breaking backwards compatibility, the exported symbol `AbstractComponent` in `@blueprintjs/core` must behave identically to the currently released one. you should make a new `AbstractComponentWithNewLifecycle` component and use that in the new components you are polyfilling.

Reply:
i've done this, although i do not think it is a good idea. it will propagate usage of the old `AbstractComponent` by external consumers and cause internal renaming work down the line.